public/app/features/dashboard/containers/DatasourceOnboarding.tsx

```diff
@@ -0,0 +1,209 @@
+
+const topDatasources = [
+  'prometheus',
+  'mysql',
+  'elasticsearch',
+  'influxdb',
+  'graphite',
+  'stackdriver',
+  'cloudwatch',
+  'grafana-azure-monitor-datasource',
+];
+
+export function DatasourceOnboarding({
+  onNewDashboard,
+  loading = false,
+}: {
+  onNewDashboard?: () => void;
+  loading?: boolean;
+}) {
+  const styles = useStyles2(getStyles);
+  const { value: datasources, loading: loadingDatasources } = useAsync(async () => {
+    const datasourceMeta: DataSourcePluginMeta[] = await getBackendSrv().get('/api/plugins', {
+      enabled: 1,
+      type: 'datasource',
+    });
+
+    const byId = datasourceMeta.reduce<Record<string, DataSourcePluginMeta>>((prev, cur) => {
+      prev[cur.id] = cur;
+      return prev;
+    }, {});
+
+    return topDatasources.map((d) => byId[d]);
+  }, []);
+
+  const onAddDatasource = useAddDatasource();
+
+  if (loading) {
+    return null;
+  }
+
+  return (
+    <Page
+      navId="dashboards/browse"
+      pageNav={{ text: t('dashboard', 'New dashboard'), url: '/dashboard/new' }}
+      layout={PageLayoutType.Canvas}
+    >
+      <div className={styles.wrapper}>
+        <h1 className={styles.title}>{t('datasource-onboarding.welcome', 'Welcome to Grafana dashboards!')}</h1>
+        <div className={styles.description}>
+          <h4 className={styles.explanation}>
+            {t('datasource-onboarding.explanation', "To visualize your data, you'll need to connect it first.")}
+          </h4>
+        </div>
+        {contextSrv.hasRole('Admin') ? (
+          <>
+            <h4 className={styles.preferredDataSource}>
+              {t('datasource-onboarding.preferred', 'Connect your preferred data source:')}
+            </h4>
+            {!loadingDatasources && datasources !== undefined && (
+              <ul className={styles.datasources}>
+                {datasources.map((d) => (
+                  <li key={d.id}>
+                    <button onClick={() => onAddDatasource(d)}>
+                      <img
+                        src={d.info.logos.small}
+                        alt={t('datasource-onboarding.logo', 'Logo for {{datasourceName}} data source', {
+                          datasourceName: d.name,
+                        })}
+                        height="16"
+                        width="16"
+                        className={styles.logo}
+                      />
+                      <span className={styles.datasourceName}>{d.name}</span>
+                      <Icon name="arrow-right" size="lg" className={styles.arrowIcon} />
+                    </button>
+                  </li>
+                ))}
+                <li>
+                  <a href="/datasources/new" className={styles.viewAll}>
+                    <span>{t('datasource-onboarding.viewAll', 'View all')}</span>
+                    <Icon name="arrow-right" size="lg" />
+                  </a>
+                </li>
+              </ul>
+            )}
```

Comment:
@kaydelaney  I've noticed a flicker when transitioning to new dashboard page due to this. See the video (slowed down intentionally):

https://user-images.githubusercontent.com/2376619/205246100-09d6f225-fd78-4883-ac55-8f0f3808c102.mp4

Maybe we can render a loading indicator until datasources are loaded, and only then render the entire page?